### PR TITLE
bears/naturallanguage: Add SpellCheckLintBear

### PR DESCRIPTION
**Pull request by mr-karan**
This bear uses ``scspell`` to check for
spelling mistakes.

Closes https://github.com/coala-analyzer/coala-bears/issues/629

### CODE REVIEW

File: bears/natural_language/SpellCheckLintBear.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from coalib.bears.requirements.PipRequirement import PipRequirement
+
+
+@linter(executable='scspell',
+        use_stderr=True,
+        output_format='regex',
+        output_regex=r'(?P<filename>.*):(?P<line>.\d*):\s*(?P<message>.*)')
+class SpellCheckLintBear:
+    """
+    Lints the files to check for incorrect spellings using ``scspell``.
```
Comment (Asnelchristian) [inline bears/natural_language/SpellCheckLintBear.py]: ``Lints files``

File: requirements.txt [hunk 2]
```diff
@@ -28,3 +28,4 @@ vim-vint==0.3.*
 nltk==3.1.*
 appdirs==1.*
 pyyaml==3.*
+scspell3k=2.*
```
Comment (Asnelchristian) [inline requirements.txt]: ``==`` not ``=``

**Comment by mr-karan** [comment]
https://asciinema.org/a/80871

**Comment by mr-karan** [comment]
I have no clue why AppVeyor fails when the same thing passes locally and on CI

File: bears/natural_language/SpellCheckLintBear.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    LANGUAGES = {"Natural Language"}
+    REQUIREMENTS = {PipRequirement('scspell3k', '2.0')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Spelling'}
+
+    @staticmethod
+    def create_arguments(filename, file, config_file):
+        args = ('--report-only', filename,)
```
Comment (arafsheikh) [inline bears/natural_language/SpellCheckLintBear.py]: Looks like `scspell` takes in a few useful arguments like dictionary and escaping (https://github.com/myint/scspell).

**Comment by arafsheikh** [comment]
Marking WIP

**Comment by AbdealiJK** [comment]
I'm taking over this in https://github.com/coala-analyzer/coala-bears/pull/747 as it's been a while since it's been updated

**Pull request closed without merging**